Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
+        // already promisified
+        Document : processDocument,
+        // already promisified
+        Folder : processFolder,
+        Placemark : deferAndWrapInPromise(processPlacemark),
+        NetworkLink : deferAndWrapInPromise(processNetworkLink),
+        GroundOverlay : deferAndWrapInPromise(processGroundOverlay),
+        PhotoOverlay : deferAndWrapInPromise(processUnsupported),
+        ScreenOverlay : deferAndWrapInPromise(processUnsupported),
+        Tour : deferAndWrapInPromise(processUnsupported)
+    };
+
+    function deferAndWrapInPromise(originalFunction){
+        return function(){
+            var deferred = when.defer();
+            var args = Array.prototype.slice.call(arguments);
+            var context = this;
+            setTimeout(function deferredFunc(){
```

Comment:
Please don't tie loading of data too closely to the Cesium render loop.  It will break applications that shut down Cesium's render loop to avoid burning 100% CPU all the time.  I don't mind if I need to do extra work in my app to make loading progress while rendering is stopped, but please think about how it's possible at least.